runtime/parser2/lexer/lexer.go

```diff
@@ -95,15 +96,35 @@ func (l *lexer) Revert(cursor int) {
 	l.cursor = cursor
 }
 
+func (l *lexer) clear() {
+	l.startOffset = 0
+	l.endOffset = 0
+	l.prevEndOffset = 0
+	l.current = EOF
+	l.prev = EOF
+	l.canBackup = false
+	l.startPos = position{line: 1}
+	l.cursor = 0
+	l.tokens = l.tokens[:0]
+	l.tokenCount = 0
+}
+
+func (l *lexer) Reclaim() {
+	pool.Put(l)
+}
+
+var pool = sync.Pool{
+	New: func() interface{} {
+		return &lexer{
+			tokens: make([]Token, 0, 2048),
```

Comment:
@turbolent sorry, got here after the PR has been merged. But left a comment here [1] suggesting that the `2048` be increased 10 fold to e.g. `20480`. Why? Already found a contract which increases the token capacity to almost 16,000...

[1] https://axiomzen.slack.com/archives/CG0B7CJAJ/p1652314319091899?thread_ts=1651116257.517069&cid=CG0B7CJAJ

Reply:
We should probably not optimize for large cases, but for the average. Otherwise we have to pay for this large memory usage for all parses, which often encounter way fewer tokens

Reply:
@turbolent Thinking out loud: What do you think about the idea (for a future PR) of limiting the total number of lexers to e.g. some multiple of the number of CPU cores on the box? So on mainnet ENs it might be e.g. 2 * 128 = 256 possible lexer instances? In theory this would limit the memory footprint for the token slices and stop the ENs from being overwhelmed by a surge of concurrent Cadence script lexing via GRPC requests? There would have to be some queuing mechanism so that if all n lexer instances are currently in use, then the n+1 wannabe user would be queued and wait for one of the existing lexer instances to finish?

Reply:
Limiting the number of concurrent executions, e.g. for scripts, sounds like a good idea. However, I would recommend implementing this at a higher level than the lexer, even higher than Cadence itself, i.e. in the FVM/EN.

Reply:
It would be good to revisit this idea after this PR hits mainnet and we can see the effect on the lexer garbage production for a while. We'll then know if further optimization is necessary to curb the lexer garbage production.

Who would be a good person to approach to implement this at the higher level? Maybe @Kay-Zee has suggestions? :-)